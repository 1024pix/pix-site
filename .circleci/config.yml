---
version: 2.0

jobs:
  test:
    docker:
      - image: cimg/node:16.20.0
    steps:
      - checkout
      - run: cat package*.json > cachekey
      - restore_cache:
          keys:
            - pix-site-npm-{{ checksum "cachekey" }}
      - run: npm ci
      - save_cache:
          key: pix-site-npm-{{ checksum "cachekey" }}
          paths:
            - ~/.npm
      - run:
          name: Test
          command: |
            npm test
      - run:
          name: Lint
          command: |
            npm run lint

  e2e:
    docker:
      - image: mcr.microsoft.com/playwright:v1.30.0-focal
    resource_class: medium
    steps:
      - checkout
      - run: cat package*.json > cachekey
      - restore_cache:
          keys:
            - v2-pix-site-e2e-npm-{{ checksum "cachekey" }}
      - run: npm ci
      - save_cache:
          key: v2-pix-site-e2e-npm-{{ checksum "cachekey" }}
          paths:
            - ~/.npm
            - ~/.cache/ms-playwright
      - run:
          name: Run E2E tests
          environment:
            PLAYWRIGHT_JUNIT_OUTPUT_NAME: test-results/e2e-junit-results.xml
          command: npm run test:e2e:run -- --reporter=junit
      - store_test_results:
          path: test-results/

workflows:
  version: 2
  build_and_test:
    jobs:
      - test
      - e2e
