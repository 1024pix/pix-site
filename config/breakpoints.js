export const DESKTOP_MIN_WIDTH = 960
export const EXTRA_LARGE_SCREEN_MIN_WIDTH = 1440
