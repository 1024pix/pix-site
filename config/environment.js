import { SITES_PRISMIC_TAGS } from '../services/available-sites'

export const config = {
  site: process.env.SITE,
  siteDomain: process.env.SITE_DOMAIN,
  isFrenchDomain: process.env.SITE_DOMAIN === 'pix.fr',
  isPixSite: process.env.SITE === SITES_PRISMIC_TAGS.PIX_SITE,
  prismic: {
    apiEndpoint: process.env.PRISMIC_API_ENDPOINT,
  },
  matomo: {
    containerUrl: process.env.MATOMO_CONTAINER,
    debug: process.env.MATOMO_DEBUG || false,
  },
  featureToggles: {
    disableLanguageSwitcherPixProFr:
      process.env.FT_DISABLE_PIX_PRO_LANGUAGE_SWITCHER === 'true' &&
      process.env.SITE === SITES_PRISMIC_TAGS.PIX_PRO &&
      process.env.SITE_DOMAIN === 'pix.fr',
  },
}
