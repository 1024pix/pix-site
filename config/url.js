/**
 * Warning, if you want to use this function in SSR,
 * it only works in dev mode due to req not being present in static generation build
 * (see {@link https://nuxtjs.org/docs/internals-glossary/context/#req})
 * @returns {string} a base url (ex: http://localhost:7000)
 */
export function getBaseUrl(context) {
  const { isDev, req } = context
  const protocol = isDev ? 'http' : 'https'

  return process.server
    ? `${protocol}://${req?.headers?.host}`
    : window.location.origin
}
