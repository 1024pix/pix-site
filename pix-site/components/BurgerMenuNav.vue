<template>
  <div class="burger-menu-nav">
    <burger-menu-nav-items :items="items.navigationZone" />
    <burger-menu-nav-items :items="items.actionsZone" :is-action="true" />
    <locale-switcher type="only-text" />
  </div>
</template>

<script>
export default {
  name: 'BurgerMenuNav',
  props: {
    items: {
      type: Object,
      default: null,
    },
  },
}
</script>

<style lang="scss">
.burger-menu {
  .bm-menu {
    background: $white;

    @include device-is('large-screen') {
      display: none;
    }
  }

  .bm-item-list {
    height: 100%;

    & > * {
      padding: 0;
      flex-direction: column;
      height: 100%;
    }
  }

  .bm-burger-bars {
    background-color: $grey-45;
  }

  .bm-burger-button {
    top: 25px;
    left: 24px;

    @include device-is('tablet') {
      left: 48px;
    }

    @include device-is('large-screen') {
      display: none;
    }
  }

  .bm-cross {
    background: $grey-45;
    height: 24px !important;
  }
}
</style>
