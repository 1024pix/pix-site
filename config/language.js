import { config } from './environment'

export const language = {
  menu: [
    {
      name: 'international',
      icon: 'globe-europe.svg',
      target: null,
      sub: null,
      children: [
        {
          name: 'french',
          lang: 'fr',
          icon: 'icon',
          target: '/fr',
          isOnPixOrg: true,
          sub: null,
        },
        {
          name: 'english',
          lang: 'en-gb',
          icon: 'icon',
          target: '/en-gb',
          isOnPixOrg: true,
          sub: null,
        },
      ],
    },
    {
      name: 'france',
      lang: 'fr-fr',
      icon: 'flag-fr.svg',
      target: '/',
      isOnPixOrg: false,
      sub: null,
    },
    {
      name: 'fwb-acronym',
      lang: 'fr-be',
      icon: 'flag-be.svg',
      target: '/fr-be',
      isOnPixOrg: true,
      sub: 'fwb',
    },
  ],
  locales: config.isFrenchDomain
    ? [
        {
          code: 'fr-fr',
          file: 'fr-fr.js',
        },
      ]
    : [
        {
          code: 'fr',
          file: 'fr.js',
        },
        {
          code: 'en-gb',
          file: 'en-gb.js',
        },
        {
          code: 'fr-be',
          file: 'fr-be.js',
        },
      ],
}
