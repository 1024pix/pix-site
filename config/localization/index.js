import { config } from '../environment'
import { localization as localizationForPixSite } from './pix-site'
import { localization as localizationForPixPro } from './pix-pro'

export const localization = config.isPixSite
  ? localizationForPixSite
  : localizationForPixPro
