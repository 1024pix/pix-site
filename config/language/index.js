import { config } from '../environment'
import { language as languageForPixSite } from '../language/pix-site'
import { language as languageForPixPro } from '../language/pix-pro'

export const language = config.isPixSite
  ? languageForPixSite
  : languageForPixPro
