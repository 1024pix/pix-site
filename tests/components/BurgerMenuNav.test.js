import { shallowMount } from '@vue/test-utils'
import BurgerMenuNav from '~/components/BurgerMenuNav'
import { config } from '~/config/environment'

describe('Component: Burger Menu Nav', () => {
  let component

  describe('When FT_DISABLE_PIX_PRO_LANGUAGE_SWITCHER is set to true', () => {
    jest.mock('~/config/environment', () => {
      return {
        config: {
          featureToggles: {
            disableLanguageSwitcherPixProOrg: true,
          },
        },
      }
    })

    it('it should not display the language switcher', () => {
      // given
      component = shallowMount(BurgerMenuNav, {
        stubs: {
          'language-switcher': true,
          'burger-menu-nav-items': true,
        },
        propsData: {
          items: {},
        },
      })

      // when
      const result = component.html()

      // then
      expect(result).toMatchSnapshot()
    })
  })

  describe('When FT_DISABLE_PIX_PRO_LANGUAGE_SWITCHER is set to false', () => {
    beforeEach(() => {
      config.featureToggles.disableLanguageSwitcherPixProOrg = false
    })

    it('it should display the language switcher', () => {
      // given
      component = shallowMount(BurgerMenuNav, {
        stubs: {
          'language-switcher': true,
          'burger-menu-nav-items': true,
        },
        propsData: {
          items: {},
        },
      })

      // when
      const result = component.html()

      // then
      expect(result).toMatchSnapshot()
    })
  })
})
