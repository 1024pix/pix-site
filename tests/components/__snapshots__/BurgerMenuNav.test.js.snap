// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Burger Menu Nav When FT_DISABLE_PIX_PRO_LANGUAGE_SWITCHER is set to false it should display the language switcher 1`] = `
"<div class=\\"burger-menu-nav\\">
  <burger-menu-nav-items-stub></burger-menu-nav-items-stub>
  <burger-menu-nav-items-stub is-action=\\"true\\"></burger-menu-nav-items-stub>
  <language-switcher-stub type=\\"only-text\\"></language-switcher-stub>
</div>"
`;

exports[`Component: Burger Menu Nav When FT_DISABLE_PIX_PRO_LANGUAGE_SWITCHER is set to true it should not display the language switcher 1`] = `
"<div class=\\"burger-menu-nav\\">
  <burger-menu-nav-items-stub></burger-menu-nav-items-stub>
  <burger-menu-nav-items-stub is-action=\\"true\\"></burger-menu-nav-items-stub>
  <language-switcher-stub type=\\"only-text\\"></language-switcher-stub>
</div>"
`;
